### PR TITLE
Raising more precise API error when set bitrate fails

### PR DESCRIPTION
**Pull request by pierreluctg**
Also calling shutdown before raising exception from the init if the device is opened

### CODE REVIEW

File: can/interfaces/ics_neovi/neovi_bus.py [hunk 1]
```diff
@@ -158,15 +158,23 @@ def __init__(self, channel, can_filters=None, **kwargs):
...
+            if fd:
+                if "data_bitrate" in kwargs:
+                    for channel in self.channels:
+                        ics.set_fd_bit_rate(
+                            self.dev, kwargs.get("data_bitrate"), channel
+                        )
+        except ics.RuntimeError as re:
+            logger.error(re)
+            err = ICSApiError(*ics.get_last_api_error(self.dev))
+            self.shutdown()
```
Comment (felixdivo) [inline can/interfaces/ics_neovi/neovi_bus.py:176]: Just a thought: can this throw an exception again? Would we also want to catch that?

File: can/interfaces/ics_neovi/neovi_bus.py [hunk 2]
```diff
@@ -158,15 +158,23 @@ def __init__(self, channel, can_filters=None, **kwargs):
...
-        fd = kwargs.get("fd", False)
-        if fd:
-            if "data_bitrate" in kwargs:
+        try:
+            if "bitrate" in kwargs:
                 for channel in self.channels:
-                    ics.set_fd_bit_rate(self.dev, kwargs.get("data_bitrate"), channel)
+                    ics.set_bit_rate(self.dev, kwargs.get("bitrate"), channel)
+
+            fd = kwargs.get("fd", False)
```
Comment (felixdivo) [inline can/interfaces/ics_neovi/neovi_bus.py:166]: Could be inlined into the if statement's condition. (Does not need to be a variable.) That's nothing too important though. ;)

Comment (pierreluctg) [reply can/interfaces/ics_neovi/neovi_bus.py:166]: Agree, but will not touch it in this PR

File: can/interfaces/ics_neovi/neovi_bus.py (hunk 1 above)
Comment (pierreluctg) [reply can/interfaces/ics_neovi/neovi_bus.py:176]: Yes, `shutdown` (which call `ics.close_device` can raise a `ics.RuntimeError`. However, in this case we do not want to call `ics.get_last_api_error`.

Will fix

**Comment by pierreluctg** [comment]
@hardbyte can you please merge. The `test_thread_based_cyclic_send_task` keep failing either on Trusty or OSX.

File: can/interfaces/ics_neovi/neovi_bus.py (hunk 1 above)
Comment (felixdivo) [reply can/interfaces/ics_neovi/neovi_bus.py]: Cool!

**Comment by felixdivo** [comment]
They randomly worked when I restarted them a minute ago.

**Pull request merged**